architect_py/async_client.py

```diff
@@ -308,6 +308,62 @@ async def send_limit_order(
         )
 
         return await self.get_order(order)
+    
+    async def send_market_pro_order(
+        self,
+        *,
+        market: str,
+        dir: OrderDirection,
+        quantity: DecimalLike,
+        trigger_price: Optional[DecimalLike] = None,
+        time_in_force: CreateTimeInForceInstruction = CreateTimeInForceInstruction.DAY,
+        account: Optional[str] = None,
+        source: OrderSource = OrderSource.API,
+        quote_id: Optional[str] = None,
+        percent_through_market: int = 2
+    ) -> Optional[GetOrderOrder]:
+
+        # Check for GQL failures
+        bbo_snapshot = await self.get_market_snapshot(market)
+        if bbo_snapshot is None:
+            raise ValueError("Failed to send market order with reason: no market snapshot for {market}")
+        
+        market_details = await self.get_market(market)
+        if market_details is None:
+            raise ValueError("Failed to send market order with reason: no market details for {market}")
+
+        best_bid = float(bbo_snapshot.bid_price)
+        best_offer = float(bbo_snapshot.ask_price)
+        last_price = float(bbo_snapshot.last_price)
+        
+        limit_price = best_offer * (1 - percent_through_market/100) if dir == OrderDirection.BUY else best_bid * (1 + percent_through_market/100)
```

Comment:
the signs are flipped. I don't think this was tested...

Reply:
Certainly not thoroughly enough. Paper trading has been down for some time - once that gets fixed I'll be able to do more thorough testing.